test/scripts/test_externalTests_benchmark_downloader.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python3
```

Comment:
I always find it admirable how commited you are to unit-testing everything ;-). But I'm not entirely sure, in this particular case, that this is worth 250 lines in the repo :-). You also don't run this anywhere, right?
I'm fine with keeping it - it may also just serve to document the helper scripts... but I also wouldn't say we really need this :-).

Reply:
> You also don't run this anywhere, right?

True, you cannot run this on the command line (it's not even executable), the shebang should not be there :) I'll remove it.

> But I'm not entirely sure, in this particular case, that this is worth 250 lines in the repo :-).

Honestly, I'd say that the amount of tests here is pretty low for a scripted language. In other projects I could rarely get away with fewer lines in tests than in the script itself :) Interpreted code is just so brittle if you don't execute everything at least once. One of the things I hate when dealing with our Bash scripts is that I often cannot be sure that my changes do not break something. This is often a problem with ones that are not even plugged into CI (like PPA build or wasm rebuilds). It was actually a big obstacle back when I was doing a mass refactor to enable shellcheck. With Python we can avoid that and I think that each script should have at least some minimal coverage.

And this **is** minimal. These aren't even proper unit tests covering each function in isolation. I'm only calling the top-level function with the each important combination of options. And only ~100 lines here are really tests. The rest are JSON fixtures of responses (which I could move to standalone files in `fixtures/` but I thought it was better to have them here).

It also did actually help me to find some subtle bugs I had. E.g. initially I was sorting it wrong and not actually choosing the latest run, which is not that easy to tell unless you actually tediously inspect all the responses. I could do more thorough manual testing but I really find it more convenient to have any tests I did in the form of a test suite I can rerun. Maybe I'm just spoiled by tests covering our C++ code. I want the same thing for scripts :)